### PR TITLE
Rapid Test Profile List Screen (EXPOSUREAPP-12606)

### PR DESCRIPTION
**Pull request by SamuraiKek**
Screen Shot 2022-04-11 at 18 05 36

Currently using dummy data. Should work once #5074 is merged

### CODE REVIEW

**Review by KathrinLuecke — DISMISSED** [review]
UA approved

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/list/ProfileListEvent.kt
Comment (chiljamgossow) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/list/ProfileListEvent.kt:6]: The id is changing to be an Int

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/list/ProfileListFragment.kt
Comment (chiljamgossow) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/list/ProfileListFragment.kt:77]: Forward the id here?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/list/items/ProfileCard.kt
Comment (chiljamgossow) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/list/items/ProfileCard.kt:29]: You need to check whether the names are bank here, otherwise it will look ugly with the needless space as a separator

Comment (chiljamgossow) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/list/items/ProfileCard.kt:29]: The profile should not be nullable in the list

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/viewmodel/SubmissionDispatcherViewModel.kt
Comment (chiljamgossow) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/viewmodel/SubmissionDispatcherViewModel.kt:35]: Remove the Rat?

File: Corona-Warn-App/src/main/res/layout/profile_card_item.xml
Comment (chiljamgossow) [inline Corona-Warn-App/src/main/res/layout/profile_card_item.xml:43]: Please use even numbers where possible, not necessary to follow designs so precisely as they are not that accurate, so 26dp
Ideally we use our standard dimens

File: Corona-Warn-App/src/main/res/layout/profile_list_fragment.xml
Comment (chiljamgossow) [inline Corona-Warn-App/src/main/res/layout/profile_list_fragment.xml:21]: The title need to be plural 'Schnelltest-Profile'

File: Corona-Warn-App/src/main/res/values-de/antigen_strings.xml
Comment (chiljamgossow) [inline Corona-Warn-App/src/main/res/values-de/antigen_strings.xml:167]: Remove the RAT. It is Rapid tests, but not only antigen, also PCR

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/list/ProfileListEvent.kt
Comment (SamuraiKek) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/list/ProfileListEvent.kt:6]: Yes, but I'm waiting to merge you PR first because the id is still a string on this branch in the `Profile` data class. We'd have a conflict if I were to change it from the data class as well.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/list/items/ProfileCard.kt
Comment (SamuraiKek) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/list/items/ProfileCard.kt:29]: Alright, I was using this `PersonProfile` data class that I found in `RATProfileQrCodeFragmentViewModel` where both the profile and the qrCode are nullable, but I can actually pass the profile and the qrCode directly to avoid checking for null here.

File: Corona-Warn-App/src/main/res/layout/profile_card_item.xml
Comment (SamuraiKek) [reply Corona-Warn-App/src/main/res/layout/profile_card_item.xml:43]: Sure, it was a copy paste from somewhere else and it ended up here too 😆

File: Corona-Warn-App/src/main/res/layout/profile_list_fragment.xml
Comment (SamuraiKek) [reply Corona-Warn-App/src/main/res/layout/profile_list_fragment.xml:21]: Aaah, I thought it was a mistake and the designers wrote Profile in English because everywhere else it's Profil 😄

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/list/ProfileListFragment.kt
Comment (SamuraiKek) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/list/ProfileListFragment.kt:77]: I would also wait until #5074 is merged because I'd need to create a navArg with the id as Int.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/list/items/ProfileCard.kt
Comment (chiljamgossow) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/list/items/ProfileCard.kt]: you can just use trim()

Comment (chiljamgossow) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/list/items/ProfileCard.kt]: and make the profile not nullable, cannot be null in that fragment, just use Profile() as a fallback

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/list/items/ProfileCard.kt
Comment (chiljamgossow) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/list/items/ProfileCard.kt:28]: "${curItem.profile.firstName} ${curItem.profile.lastName}".trim()
will do it

**Review by chiljamgossow — COMMENTED** [review]
Nice!

**Comment by jurajkusnier** [comment]
I get screenshot with missing QR code when running `RATProfileQrCodeFragmentTest`
image

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/list/items/ProfileCard.kt
Comment (jurajkusnier) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/list/items/ProfileCard.kt:27]:
```suggestion
        val fullName = "${curItem.profile.firstName.trim()} ${curItem.profile.lastName.trim()}".trim()
```

File: Corona-Warn-App/src/main/res/layout/profile_list_fragment.xml
Comment (jurajkusnier) [inline Corona-Warn-App/src/main/res/layout/profile_list_fragment.xml:57]:
```suggestion
        style="@style/subtitleMediumBold"
```

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/list/ProfileListViewModel.kt
Comment (jurajkusnier) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/list/ProfileListViewModel.kt:27]: according to the documentation
> The list shall be sorted by `First Name` and `Last Name`.

Comment (jurajkusnier) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/list/ProfileListViewModel.kt:27]:
```suggestion
            profiles.sortedWith(compareBy({ it.firstName }, { it.lastName }))
```

**Review by jurajkusnier — APPROVED** [review]

**Review by chiljamgossow — APPROVED** [review]
Let's merge

**Review by KathrinLuecke — APPROVED** [review]
UA approved

**Pull request merged**
